File: javascript/hamming/hamming.js
function Hamming () {}

Hamming.prototype.compute = function (left, right) {
  if (left.length !== right.length) {
    throw new Error('DNA strands must be of equal length.')
  }
  
  var count = 0
  
  left
    .split('')
    .forEach(function (value, index) {
      if (value !== right[index]) count++
    })
  
  return count
}

module.exports = Hamming

File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

class Hamming
  
  def self.compute(left, right)
    if left.length != right.length
      raise ArgumentError
    end
    
    diff_count = 0
    
    left
      .split('')
      .each_index { |i|
        diff_count += 1 if left[i] != right[i]
      }
    
    diff_count
  end
  
end